test/spec/adapters/thoughtleadr_spec.js

```diff
@@ -0,0 +1,176 @@
+    });
+  });
+
+  describe("handleBids", function () {
+
+    beforeEach(function () {
+      tldrRequestPrebid = setupResponse({});
+    });
+
+    it("should filter invalid bids", function () {
+      request.bids.unshift({
+        bidder: "thoughtleadr",
+        placementCode: "abc-123",
+        sizes: [[300, 250], [400, 400]],
+        params: {}
+      });
+      request.bids.push({
+        bidder: "thoughtleadr",
+        placementCode: "abc-123",
+        sizes: [[300, 250], [400, 400]],
+        params: {
+          incorrectParam: 123
+        }
+      });
+      var requestPlacement = sinon.spy(adapter, "requestPlacement");
+      adapter.callBids(request);
+      chai_1.expect(requestPlacement.callCount).to.be.equal(1);
+      chai_1.expect(requestPlacement.getCall(0).args[0]).to.be.equal(request.bids[1]);
+    });
+  });
+
+  describe("requestPlacement", function () {
+
+    beforeEach(function () {
+      tldrRequestPrebid = setupResponse({
+        config: {
+          abc: 567
+        },
+        bid: {
+          code: 1,
+          cpm: 12,
+          ad: "asd"
+        }
+      });
+    });
+
+    it("should made request through page.js api", function () {
+      adapter.callBids(request);
+      chai_1.expect(tldrRequestPrebid.callCount).to.be.equal(1);
+      chai_1.expect(tldrRequestPrebid.firstCall.args[0]).to.be.equal(request.bids[0].params.placementId);
+      chai_1.expect(tldrRequestPrebid.firstCall.args[1]).to.be.length(36);
+    });
+
+    it("should call bidfactory.createBid with code 1 if ad is ok", function () {
+      var bid = request.bids[0];
+      adapter.requestPlacement(bid);
+      chai_1.expect(createBid.callCount).to.be.equal(1);
+      chai_1.expect(createBid.firstCall.args[0]).to.be.equal(1);
+    });
+
+    it("should call bidfactory.createBid with different code if ad isn't ok", function () {
+      var bid = request.bids[0];
+      tldrRequestPrebid = setupResponse({
+        config: {},
+        bid: {
+          code: 2
+        }
+      });
+      adapter.requestPlacement(bid);
+      chai_1.expect(createBid.callCount).to.be.equal(1);
+      chai_1.expect(createBid.firstCall.args[0]).to.be.equal(2);
+    });
+
+    it("should response on the postMessage request", function (done) {
+      var bid = request.bids[0];
+      adapter.requestPlacement(bid);
+      var rid = tldrRequestPrebid.firstCall.args[1];
+      chai_1.expect(rid).to.be.ok;
+
+      window.addEventListener("message", function (ev) {
+        if (ev.data && ev.data.TLDR_RESPONSE) {
+          chai_1.expect(ev.data.TLDR_RESPONSE.rid).to.be.equal(rid);
+          chai_1.expect(JSON.stringify(ev.data.TLDR_RESPONSE.config)).to.be.equal('{"abc":567}');
+          done();
+        }
+        else if (ev.data && ev.data.TLDR_REQUEST) {
+          chai_1.expect(ev.data.TLDR_REQUEST.rid).to.be.equal(rid);
+        }
+        else {
+          throw new Error("should not be any other messages");
```

Comment:
Running tests against IE9 trips up here. I haven't dug deep enough to find a more elegant way of fixing it, but changing the line to done() makes the test pass across browsers, though it may not be what you want. Are you able to run tests on IE9 and/or know of a cross-browser way of addressing this?